src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from "react"
+import { useParams } from "react-router-dom"
+import { getArticlebyId } from "../utils/api_funcs"
+
+const SingleArticle = () => {
+    const [selectedArticle, setSelectedArticle] = useState({})
+    const {article_id} = useParams()
+     const [isLoading, setIsLoading] = useState(true)
+ 
+    useEffect(()=> {
+        getArticlebyId(article_id).then((response) => {setSelectedArticle(response.article); setIsLoading(false)})
+    })
```

Comment:
This is missing the dependency array - its currently constantly doing this request

Reply:
Its also just missing some error handling for if I got to an id that isnt there. 
At the moment, it just shows a constant loading state